src/incflo_advance.cpp

```diff
@@ -42,9 +43,19 @@ void incflo::Advance()
         }
     }
 
-    // fixme this has some specific abl stuff in it that will break for non-abl cases
-    // make this generic so plane averages can be done without abl
-    if(m_probtype==35) spatially_average_quantities_down(true);
+   if(m_plane_averaging){
+       const int axis=2;
+       PlaneAveraging *pa = new PlaneAveraging(*this, get_velocity_new(), get_tracer_new(), axis);
```

Comment:
Why is this a pointer? And where is this pointer deallocated? Can you confirm there is no memory leak here? 

```C++
PlaneAveraging pa(*this, get_velocity_new(), get_tracer_new(), axis);
````

Reply:
nice catch, thank you